src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -15,8 +15,16 @@ public class GraphTypesLookup
         private readonly object _lock = new object();
         private bool _sealed;
 
-        public GraphTypesLookup()
+        public GraphTypesLookup() : this(CamelCaseNameConverter.Instance) { }
+
+        public GraphTypesLookup(INameConverter nameConverter)
         {
+            if (nameConverter == null)
+                throw new ArgumentNullException(nameof(nameConverter));
+
+            // introspection fields use camelcase
+            NameConverter = CamelCaseNameConverter.Instance;
```

Comment:
Assigning the `NameConverter` twice is a little confusing when first looking at it.  Is this needed now with the `applyNameConverter` check?  If so, can you add a comment here why it is being assigned twice and maybe add it closer to the introspection types - `__EnumValue`, `__InputValue`, etc.

Reply:
It always did that before, but it was hidden:  When the constructor executed, the NameConverter was set to the property default (CamelCaseNameConverter), and then the `Create` function would set the NameConverter after the fact.  I simply changed it so that it was obvious what was happening.

As it is now, the NameConverter must be CamelCaseNameConverter while adding the introspection types, as some places `Field(x => x.Name)` is run, so the field name is pascal case.

The `applyNameConverter` check is only set to `false` when adding the three introspection fields (as opposed to types) -- `__schema`, `__type`, and `__typename` -- on the base query type.  There's a whole boatload of comments in `HandleField` to explain this.

As I mentioned before, for better or worse, this is an old design.  I have no problem updating it, but I really didn't change the logic from before.  It was pretty hard to understand what was going on before, so hopefully this is at least an improvement.  Let me know if you think I should make further changes.  I did move the two NameConverter assignments immediately around the introspection types, as requested.  Let me know if you think that's clear.

Reply:
With the last change, `NameConverter` is null while adding the scalar types in the `GraphTypeLookup` constructor.  This does not matter as they do not have fields.  If there was a call to `HandleField` at that time, an `NullReferenceException` would occur.  This is probably for the best.

Reply:
Just FYI, the old code passed `typeof(SchemaIntrospection)` as the `parentType` parameter to `HandleField`, which would then check the `parentType` field against `typeof(SchemaIntrospection)` to determine if it should ignore the `NameConverter`.  I simply changed it to an appropriately named boolean, to make it more clear the purpose of the code.

Reply:
Well, I added the boolean, but that's what it did before.  I'm also using `IGraphType` instead of a graph `Type` so that we don't incur a `GetType()` lookup in `GetArgument` and `HasArgument`.

Reply:
I had the boolean named `introspectionField` but with comments from @sungam3r I ended up changing it to more precisely indicate its function.  So now it is called `applyNameConverter`.